Убрал все классы-инициализаторы

ArtomStatsenko opened the pull request:
Оптимизировал код, непонятный момент с точкой спавна пули, надо разбираться...

ArtomStatsenko committed:
Убрал ShipInitializer

ArtomStatsenko committed:
Убрал инициализатор пуль

Kastel-Mihalson commented on Assets/Asteroids/Scripts/Controllers/BulletController.cs:
```diff
@@ -4,15 +4,27 @@
     {
-        _model = model;
-        _rigidbody = view.Rigidbody;
+        _model = new BulletModel(data);
```
Так как инициализация view происходит в Init, то и модель должна быть там же инициализирована. Иначе снова получается, что создаётся модель один раз, а должно при каждом вызове Init.

ArtomStatsenko replied:
А нам нужно, чтобы у каждой пули была своя модель, или мы создаем одну модель для всех пуль?

ArtomStatsenko replied:
ну хотя ладно, давай для каждой пули, чтобы если будет меняться какой-то параметр у каждой пули (отскакивающие пульки может быть), то не было багов

Kastel-Mihalson replied:
Да, для каждой пули своя модель именно по этой причине.

ArtomStatsenko replied:
исправлю

Kastel-Mihalson commented on Assets/Asteroids/Scripts/Controllers/ShipController.cs:
```diff
@@ -3,17 +3,55 @@
-        _model = model;
-        _rigidBody = view.Rigidbody;
-        _gameObject = view.gameObject;
+        _model = new ShipModel(data);
```
Хоть корабль пока один, думаю тут тоже нужно перенести инициализацию модели из конструктора в метод инит.

ArtomStatsenko replied:
хорошо, согласен

Kastel-Mihalson commented on Assets/Asteroids/Scripts/Controllers/ShootingController.cs:
```diff
@@ -3,17 +3,19 @@
     private LayerMask _enemyMask;
     private Transform _startPoint;
-    private BulletInitializer _bulletInitializer;
+    private float _shootDelay;
```
Инициализацию _shootingDistance может в bulletData перенести и оттуда брать данные?

ArtomStatsenko replied:
я его сюда вынес, потому что это параметр по идее не зависит от пуль, это скорее расстояние до края экрана, которое рэйкастит корабль.

ArtomStatsenko committed:
Экземпляры моделей создаются в Init

1. Перенес создание экземпляров Model в методы Init
2. Мелкие фиксы с параметрами и названиями переменных
